packages/client/bin/repl.ts

```diff
@@ -0,0 +1,85 @@
+import process from 'process'
+import repl from 'repl'
+
+const setupClient = async () => {
+  const { readFileSync } = await import('fs')
+
+  const { createCommonFromGethGenesis } = await import('@ethereumjs/common')
+  const { createInlineClient } = await import('../test/sim/simutils.ts')
+  const { Config } = await import('../src/config.ts')
+  const { getLogger } = await import('../src/logging.ts')
+  const { startRPCServers } = await import('./startRPC.ts')
+  const genesisFile = JSON.parse(readFileSync('./bin/genesis.json', 'utf-8'))
+  const chainName = 'pectra'
+  const common = createCommonFromGethGenesis(genesisFile, {
+    chain: chainName,
+  })
+  const config = new Config({
+    common,
+    logger: getLogger({ logLevel: 'info' }),
+    saveReceipts: true,
+    enableSnapSync: true,
+  })
+  const client = await createInlineClient(config, common, {}, '', true)
+  const servers = startRPCServers(client, {
+    rpc: true,
+    rpcAddr: '0.0.0.0',
+    rpcPort: 8545,
+    ws: false,
+    wsPort: 0,
+    wsAddr: '0.0.0.0',
+    rpcEngine: true,
+    rpcEngineAddr: '0.0.0.0',
+    rpcEnginePort: 8551,
+    wsEngineAddr: '0.0.0.0',
+    wsEnginePort: 8552,
+    rpcDebug: 'eth',
+    rpcDebugVerbose: 'false',
+    helpRPC: false,
+    jwtSecret: '',
+    rpcEngineAuth: false,
+    rpcCors: '',
+  })
+  return { client, executionRpc: servers[0], engineRpc: servers[1] }
+}
+
+const setupRepl = async () => {
+  const { client, executionRpc, engineRpc } = await setupClient()
+
+  const replServer = repl.start({
+    prompt: 'EthJS > ',
+    ignoreUndefined: true,
+  })
+
+  // bootstrap contexts or modules
+  replServer.context.client = client
+  replServer.context.executionRpc = executionRpc._methods // TODO modify methods to only include functions and make them usable
+  replServer.context.engineRpc = engineRpc._methods
+
+  replServer.on('exit', () => {
+    console.log('Exiting REPL...')
+    process.exit()
+  })
+
+  // define commands
```

Comment:
Commands can be defined as such that would allow use to simplify rpc API function calls and passing input parameters. Any other processing can also be done in the command callback implementation, but we should be careful not to e.g. reimplement rpc functions here.